### ISSUE TITLE
mod global is getting "lost"

### ISSUE THREAD

**Issue by muppet9010** [JanSharp/FactorioSumnekoLuaPlugin#4]
I have issues where my Factorio `global` of my mod is becoming just a lua global called `global` in various points of my mods files.

JustARandomGeek suggested I used the `--develop=true` parameter to get the `diffed` file that shows the issue.

the snippet with the correct mod `global` and then the generic global is below:
```lua
LeakyFlamethrower.CreateGlobals = function()
    __factorio_muppet_streamer__global.leakyFlamethrower = __factorio_muppet_streamer__global.leakyFlamethrower or {}
    __factorio_muppet_streamer__global.leakyFlamethrower.affectedPlayers = __factorio_muppet_streamer__global.leakyFlamethrower.affectedPlayers or {} ---@type table<uint, LeakyFlamethrower_AffectedPlayersDetails> @ Key'd by player_index.
    global.leakyFlamethrower.nextId = __factorio_muppet_streamer__global.leakyFlamethrower.nextId or 0 ---@type uint
end
 ```
 In this case the globals are all written as `global.leakyFlamethrower` in the source code.

The full diffed file and the source file are attached as .txts.
[leakyflamethrower-diffed.lua.txt](https://github.com/JanSharp/FactorioSumnekoLuaPlugin/files/9155266/leakyflamethrower-diffed.lua.txt)
[leaky-flamethrower-source.lua.txt](https://github.com/JanSharp/FactorioSumnekoLuaPlugin/files/9155267/leaky-flamethrower-source.lua.txt)

**Comment by JanSharp** [comment]
That's a fun one. The `.` (dot) at the end of the line in the comment made the plugin think you're indexing using `.global` so it didn't touch it.

I've fixed that, let me know if that fixes all of your issues or not

**Comment by muppet9010** [comment]
Its fixed the main issue across most of the files :)

However it has exposed a the below edge case. The global handling breaks in this file when I load the project in VSCode, but if I change the file in any way then it all becomes happy again.

Towards the top of the code is this odd WIP text block that I wondered if was causing it, but I removed the @type and TODO and it still complained on load.
```lua
    global.PantsOnFire.playerSteps = global.PantsOnFire.playerSteps or {} ---@type table<uint, table<>> --TODO
```

![image](https://user-images.githubusercontent.com/40836017/180223235-129b1e18-4f49-4814-ba37-f6dd49281cc6.png)
[pants-on-fire.lua.txt](https://github.com/JanSharp/FactorioSumnekoLuaPlugin/files/9159646/pants-on-fire.lua.txt)

I can't think of a way of getting the diffed file for this, as its not the last file loaded in the mod and so it isn't the diffed file at the end of starting VSCode. And as any change to the file "clears" the errors I can't use this to make it re-generate the diffed file like I could for the others.

**Comment by JanSharp** [comment]
This unfortunately sounds like it's a reoccurring problem with the way sumneko.lua handles plugins in `3.x` in general. We (Nexela, raiguard and myself) have all experienced this behavior with `require("__modname__.foo")` where it won't find the file until you modify the current file in any way. raiguard and I have determined that it is a race condition in the initial loading process of a file in sumneko.lua as it only happens like 3/4 times, not every time. The chance of it happening may depend on several factors so there's no guarantee that 3/4 is accurate for you.

So long story short there is nothing we can do about this right now. However since at this point it's 4 people experiencing the same thing it's probably about time we report it as an issue with sumneko.lua. I've just been too lazy to setup an as-small-as-possible reproduction case but I'll try my best to remember to do it today

**Comment by muppet9010** [comment]
Ah found another single one that never works correctly. Changing the file fixes any others on the page that don't behave.

On line 57 of diffed, line 56 in player-alerts.lua.txt
[diffed.lua.txt](https://github.com/JanSharp/FactorioSumnekoLuaPlugin/files/9174578/diffed.lua.txt)
[player-alerts.lua.txt](https://github.com/JanSharp/FactorioSumnekoLuaPlugin/files/9174579/player-alerts.lua.txt)

It looks to occur when I add a `.` before the `global`, in the `..`.

**Comment by JanSharp** [comment]
Damn it, didn't think of that. Fixed :)

**Comment by muppet9010** [comment]
yup it seems happy now :)
